File: Baekjoon/Implementation/[4358]생태학/solution.js
const fileName = process.platform === "linux" ? "/dev/stdin" : "example.dat";
const inputData = require("fs")
  .readFileSync(fileName)
  .toString()
  .trim()
  .split("\n");

const trees = {};
let tree = "";
let cnt = 0;

inputData.map((tree) => {
  trees[tree] = (trees[tree] || 0) + 1;
  cnt++;
});

console.log(
  Object.keys(trees)
    .sort()
    .map((key) => `${key} ${((trees[key] / cnt) * 100).toFixed(4)}`)
    .join("\n")
);

File: Baekjoon/Implementation/[4358]생태학/solution.py
import sys

trees = dict()
cnt = 0

for line in sys.stdin:
    tree = line.rstrip()
    if tree in trees:
        trees[tree] += 1
    else:
        trees[tree] = 1
    cnt += 1

for tree in sorted(trees.keys()):
    sys.stdout.write(f"{tree} {trees[tree]/cnt*100:0.4f}\n")
